cmd/build/v2/build.go

```diff
@@ -207,38 +207,43 @@ func (bc *OktetoBuilder) Build(ctx context.Context, options *types.BuildOptions)
 
 			repoHashDurationStart := time.Now()
 
-			meta.RepoHash = bc.hasher.hashProjectCommit(buildSvcInfo)
+			repoHash, err := bc.smartBuildCtrl.GetProjectHash(buildSvcInfo)
+			if err != nil {
+				bc.ioCtrl.Logger().Infof("error getting project commit hash: %s", err)
+			}
+			meta.RepoHash = repoHash
 			meta.RepoHashDuration = time.Since(repoHashDurationStart)
 
 			buildContextHashDurationStart := time.Now()
-			meta.BuildContextHash = bc.hasher.hashBuildContext(buildSvcInfo)
+
+			serviceHash, err := bc.smartBuildCtrl.GetServiceHash(buildSvcInfo)
+			if err != nil {
+				bc.ioCtrl.Logger().Infof("error getting service commit hash: %s", err)
+			}
+			meta.BuildContextHash = serviceHash
 			meta.BuildContextHashDuration = time.Since(buildContextHashDurationStart)
 
 			// We only check that the image is built in the global registry if the noCache option is not set
-			if !options.NoCache && bc.Config.IsCleanProject() && bc.Config.IsSmartBuildsEnabled() {
-
-				imageChecker := getImageChecker(buildSvcInfo, bc.Config, bc.Registry, bc.ioCtrl.Logger())
+			if !options.NoCache && bc.smartBuildCtrl.IsEnabled() {
+				imageChecker := getImageChecker(buildSvcInfo, bc.Config, bc.Registry, bc.smartBuildCtrl, bc.ioCtrl.Logger())
 				cacheHitDurationStart := time.Now()
-				buildHash := bc.hasher.hashService(buildSvcInfo)
+
+				buildHash, err := bc.smartBuildCtrl.GetBuildHash(buildSvcInfo)
+				if err != nil {
+					bc.ioCtrl.Logger().Infof("error getting build hash: %s", err)
+				}
 				imageWithDigest, isBuilt := imageChecker.checkIfBuildHashIsBuilt(options.Manifest.Name, svcToBuild, buildHash)
 
 				meta.CacheHit = isBuilt
 				meta.CacheHitDuration = time.Since(cacheHitDurationStart)
 
 				if isBuilt {
-					bc.ioCtrl.Out().Infof("Skipping build of '%s' image because it's already built for commit %s", svcToBuild, bc.hasher.GetCommitHash(buildSvcInfo))
-					// if the built image belongs to global registry we clone it to the dev registry
-					// so that in can be used in dev containers (i.e. okteto up)
-					if bc.Registry.IsGlobalRegistry(imageWithDigest) {
-						bc.ioCtrl.Logger().Debugf("Copying image '%s' from global to personal registry", svcToBuild)
```

Comment:
why we have removed this condition?

Reply:
We moved into the smart build package. We agreed that everything related to the smart build should be in its own package